Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -113,9 +113,6 @@ public async Task Push(Layer layer, string name, Action<string> logProgressMessa
 
         Debug.Assert(pushResponse.StatusCode == HttpStatusCode.Accepted);
 
-        //Uri uploadUri = new(BaseUri, pushResponse.Headers.GetValues("location").Single() + $"?digest={layer.Descriptor.Digest}");
-        Debug.Assert(pushResponse.Headers.Location is not null);
```

Comment:
This was confusing nullability analysis later, but the check only applied on debug builds so we hit a NRE on retail.